.travis.yml

```diff
@@ -57,8 +59,27 @@ matrix:
     - compiler: ghc-head
     - compiler: ghc-8.8.1
 before_install:
+  - |
+    if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+      if $GHCJS ; then sudo add-apt-repository -y ppa:hvr/ghcjs ; fi;
+      if $GHCJS ; then curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add - ; fi;
+      if $GHCJS ; then sudo apt-add-repository 'https://deb.nodesource.com/node_8.x xenial main' ; fi;
+      sudo apt-get update;
+      sudo apt-get install $CC freeglut3-dev;
+      if $GHCJS ; then sudo apt-get install -y nodejs cabal-install-3.0 ; fi;
+    fi
```

Comment:
Apologies for focusing so much on this part, but this still seems wrong to me. I would have expected this part of the script to (1) only run if `$GHCJS` is set, and (2) only download `ghcjs` and `nodejs`, since those are the only packages that we can't list in `addons`. AFAICT, (1) isn't happening because this happens on any Linux builder (which means that we needlessly call `apt-get update` on non-GHCJS configurations), and (2) isn't happening because we install things besides `ghcjs`/`nodejs`.

This extra work can add up in the total build time. For instance, it takes 54.77s to run all of this on the GHC 8.6 configuration, as shown [here](https://travis-ci.org/ekmett/lens/jobs/572064517#L377).

Why can't we just do this?

```
  - |
    if [ "$TRAVIS_OS_NAME" = "linux" && $GHCJS ]; then
      sudo add-apt-repository -y ppa:hvr/ghcjs;
      curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add -;
      sudo apt-add-repository 'https://deb.nodesource.com/node_8.x xenial main';
      sudo apt-get update;
      sudo apt-get install -y $CC nodejs;
    fi
```

Reply:
This step takes <1s on a non GHCJS build now.